ambry-network/src/main/java/com.github.ambry.network/NetworkMetrics.java

```diff
@@ -44,7 +44,8 @@
   public final Counter selectorKeyOperationErrorCount;
   public final Counter selectorCloseKeyErrorCount;
   public final Counter selectorCloseSocketErrorCount;
-  public Gauge<Long> selectorActiveConnections;
+  public Gauge<Long> numActiveConnections;
+  public Gauge<Long> numConnectionsPendingHandshake;
```

Comment:
This is just to track the pool (pending connections) size. This shouldn't be linearly growing. I expect the number to be stable as each connections take 5 to 6 poll() calls to complete its handshake. If the pending connections count is growing (which is not feasible to track if not for this metric), for some reason, we are doing something wrong. But that being said, I am not very pertinent in having this metric. @vgkholla @xiahome whats your take on this.